app/modules/collaborations/models.py

```diff
@@ -17,13 +51,161 @@ class Collaboration(db.Model, HoustonModel):
+                # If you initiate you approve
+                log.warning(' setting APPROVED!')
+                collab_user_assoc.read_approval_state = CollaborationUserState.APPROVED
+
+            # with db.session.begin():
+            #     db.session.add(collab_user_assoc)
+            #     db.session.merge(user_tup[0])
+
+    def get_users(self):
+        users = []
+        for association in self.collaboration_user_associations:
+            users.append(association.user)
+        return users
+
+    def get_initiators(self):
+        users = []
+        for association in self.collaboration_user_associations:
+            if association.initiator is True:
+                users.append(association.user)
+        return users
+
+    def get_read_state(self):
+        read_state = None  # if you return this the collaboration is corrupt
+        for association in self.collaboration_user_associations:
+            if association.read_approval_state == CollaborationUserState.DECLINED:
+                # only one user need decline
+                read_state = CollaborationUserState.DECLINED
+                break
+            elif association.read_approval_state == CollaborationUserState.PENDING:
+                # only one user needs pending, but we should still keep looking for declined
+                read_state = CollaborationUserState.PENDING
+            elif (
+                association.read_approval_state == CollaborationUserState.APPROVED
+                and read_state is None
+            ):
+                # every association must approve to return approved, everything else overrides it
+                read_state = CollaborationUserState.APPROVED
+        return read_state
+
+    def get_edit_state(self):
+        edit_state = None
+        for association in self.collaboration_user_associations:
+            # you should never have a 'pending' and 'not_initiated' on the same collab, so one of either is enough
+            if association.edit_approval_state == CollaborationUserState.DECLINED:
+                edit_state = CollaborationUserState.DECLINED
+                break
+            elif association.edit_approval_state == CollaborationUserState.NOT_INITIATED:
+                edit_state = CollaborationUserState.NOT_INITIATED
+                break
+            elif association.edit_approval_state == CollaborationUserState.PENDING:
+                edit_state = CollaborationUserState.PENDING
+            elif (
+                association.read_approval_state == CollaborationUserState.APPROVED
+                and edit_state is None
+            ):
+                edit_state = CollaborationUserState.APPROVED
+        return edit_state
+
+    def set_read_approval_state_for_user(self, user_guid, state):
+        from app.modules.users.models import User
+
+        if (
+            user_guid is not None
+            and User.ensure_edm_obj(user_guid) is not None
```

Comment:
is the ensure_edm_user stuff needed? I thought we were going with Jon's solution of if we have a guid, we have sufficient of the thing to use it in houston?

Reply:
Removing this any place it's just used as a check. I think it's required in the **collab_user_assoc** constructor.